lib/rgeo/cartesian/planar_graph.rb

```diff
@@ -166,17 +166,15 @@ def intersection_map
           # list of intersections for that edge.
           s1_intersects = int.point != int.s1.s && int.point != int.s1.e
           if s1_intersects
-            unless intersection_map[int.s1]
-              intersection_map[int.s1] = []
-            end
+            intersection_map[int.s1] = [] unless intersection_map[int.s1]
```

Comment:
non-blocking

```ruby
intersection_map[int.s1] ||= []
```

It seems to me that those two could be written that way

Reply:
I did have it that way, but then I had issue with tests passing (see https://github.com/rgeo/rgeo/pull/338#issuecomment-1293216948). Since I don't know the internals nor the tests very well I wasn't sure what would be causing this to happen because to me it is the same, but I'm not sure if there is a `.nil?` check somewhere that should be an `.empty?` & `.nil?` / `.blank?` check or what.

Reply:
So in the commit you mention (https://github.com/thestelz/rgeo/commit/60deae0ec466f144c29d38acf7d71617e68d7cf6), the only culprit was actually the last one. The others are ok.

It is often easier to understand `a ||= b` as `a = a || b`, which is fine.


However, the line below is not fine, because `!geom.empty?` gives `false`.

```ruby
hedge = !geom.empty? && @incident_edges[geom.start_point.coordinates].first
```

Reply:
That makes more sense to me. I guess I was just lazy and got it working so I didn't try multiple different things. I'll get em' updated!